Automattic/jetpack
# Search: Use Blog ID in links to WPCOM instead of site slug

## Proposed changes:
Adjust checkout links in Search package to use blog ID instead of slug for multi-URL support.

### Other information:


## Jetpack product discussion
pf5801-93-p2

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
To test the blog, you'll need two domains pointing to the same site. One way to do that is to setup two proxying tunnels to the same local environment.

**Connect Jetpack with a free plan on URL A, and load it via URL B.**

If your site falls into IDC due to concurrency issues with IDC secret validation (fix pending), remove `jetpack_sync_error_idc` option.

1. Using URL B (not cache site main URL), go to "Jetpack -> Search" and choose the free plan. Confirm you're redirected to the proper checkout with URL A (cache site main URL) in the page URL.
2. After purchasing, you'll end up on the Calypso "My Plan" page instead of being redirected back to the site. That's a bug to be fixed separately.
3. Go back to URL B's "Jetpack -> Search" section, confirm you see the dashboard now.
4. Click on "Upgrade Jetpack Search" and confirm you're taken to the checkout with URL A in page URL.
5. Finish the checkout. Confirm that you got redirected to URL A's "Jetpack -> Search". Confirm the plan got upgraded.
